test/driver.js

```diff
@@ -143,89 +144,79 @@ async function resolveImages(node, silentErrors = false) {
   await Promise.all(loadedPromises);
 }
 
-/**
- * @class
- */
-var rasterizeTextLayer = (function rasterizeTextLayerClosure() {
+async function rasterizeTextLayer(
+  ctx,
+  viewport,
+  textContent,
+  enhanceTextSelection
+) {
   const styles = {
     common: {
       file: "./text_layer_test.css",
       promise: null,
     },
   };
```

Comment:
Won't this re-factoring mean that you're now re-initializing the style `promise` for *every* single `rasterizeTextLayer` call, rather than caching it as intended here!?

This probably applies elsewhere below in the file as well.

Reply:
How about e.g. adding a class with static methods, e.g. something like this since that'd allow you to handle the style-caching internally instead:
```javascript
class Rasterize {
  static #textStyles = {...};

  static #annotationStyles = {...};

  static #xfaStyles = {...};

  static async textLayer(...) {}

  static async annotationLayer(...) {}

  static async xfaLayer(...) {}
}
```

Reply:
Ah, you're right. Somehow I managed to completely overlook this. I'll revisit the patch series today.

Reply:
Fixed in the new commit series.